library.js

```diff
@@ -182,23 +182,31 @@ Solr.search = function (data, callback) {
 
 		const query = Solr.client.createQuery().q(term).edismax().qf(fields).start(0).rows(500);
 
-		Solr.client.search(query, function (err, obj) {
+		Solr.ping(function(err) {
```

Comment:
Hey, can you maybe do `await Solr.ping()` instead? You'd have to promisify `Solr.ping`, or alternatively rewrite it as an asynchronous function :thinking:

Reply:
@yoyoyohamapi Solr ping is already an asynchronous function: https://github.com/julianlam/nodebb-plugin-solr/blob/bd68f72e00bcd19c36e7d5e22fffffcb8685f64d/library.js#L71

Reply:
@julianlam any idea of this?

Reply:
I mean you'd have to write it as an `async function`. Don't worry, you can leave it alone then.